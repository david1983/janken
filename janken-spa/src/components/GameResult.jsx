import React from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import GameClass from '../state/Game';

const GameResult = ({ Game }) => (
  <div className="game-result">
      <h2>We have a WINNER!!</h2>
      <h3>Player is the new EMPEROR!</h3>
      <button type="button">Play again</button>
    </div>
);

GameView.defaultProps = {
  Game: new GameClass(),
};

GameView.propTypes = {
  Game: PropTypes.object,
};


export default inject('Game')(observer(GameView));
