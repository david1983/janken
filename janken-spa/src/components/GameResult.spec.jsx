import React from 'react';
import { render } from 'enzyme';
import GameResult from './GameResult';
import GameClass from '../state/Game';

const Game = new GameClass();

describe('the GameResult component', () => {
  it('should match the snapshot of the shallow rendered component', () => {
    const component = <GameResult Game={Game} />;
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
  });
});
