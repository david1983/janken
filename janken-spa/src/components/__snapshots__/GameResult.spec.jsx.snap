// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GameResult component should match the snapshot of the shallow rendered component 1`] = `
<div
  class="game-result"
>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      />
    </div>
    <div
      class="MuiCardContent-root"
    >
      <div>
        <h1>
          We have a WINNER!!
        </h1>
        <h3>
           is the new EMPEROR!
        </h3>
      </div>
    </div>
    <div
      class="MuiCardActions-root action-area MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Play again
        </span>
      </button>
    </div>
  </div>
</div>
`;
